wordpress-mobile/WordPress-Android
# [Jetpack Social] Implement Twitter deprecation notice container on Sharing screen

Do not merge until https://github.com/wordpress-mobile/WordPress-Android/pull/18487 is merged into this branch

Fixes #18429 

<p float="left">
<img width="300" alt="Screen Shot 2023-05-22 at 10 02 03 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/c96e0e58-6dee-4bf7-9a26-e353eba1ac91">
<img width="300" alt="Screen Shot 2023-05-22 at 10 05 19 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/c3ff3e62-7c8c-4b7a-af96-44473803dda4">
</p>


To test:
1 - Install JP app and log in;
2 - Select a site that doesn't have Twitter connected;
3 - Open `Sharing` screen;
4 - Observe the services list: Twitter should not be listed;
5 - Select a site that has Twitter connected;
6 - Open `Sharing` screen again;
7 - Observe the services list: Twitter should not be listed, but Twitter deprecation container should be shown below 
8 - Tap on `Find out more` button: it should open a WebView with a post called `Why Twitter Auto-Sharing is Coming to an End`;
9 - Go back to `Sharing` screen;
10 - Tap on the Twitter row: it should open the connection details screen;
11 - Repeat the same with WP app.

## Regression Notes
1. Potential unintended areas of impact
`Sharing` screen UI

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated unit tests 

3. What automated tests I added (or what prevented me from doing so)
Updated `PublicizeListViewModelTest`, but most of it isn't testable as it is because of the existing structure of services and connections fetching.

PR submission checklist:


UI Changes testing checklist:
